src/context/TransactionContext.jsx

```diff
@@ -54,6 +55,28 @@ export const TransactionsProvider = ({ children }) => {
         }
     };
 
+    const getAllPuppies = async () => {
+        try {
+            if (ethereum) {
+            
+                const availablePuppies = await contract.getAllPuppys();
+
+                const structuredPuppies = availablePuppies.map((puppy) => ({
+                    dogId: puppy.dogId,
+                    imageUrl: puppy.imageUrl
+                }));
```

Comment:
```suggestion
```

Suggest removing `structuredPuppies` and directly use `availablePuppies` since this `map` not change anything.